br/pkg/lightning/backend/external/file.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package external
+
+import (
+	"context"
+	"encoding/binary"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pingcap/tidb/br/pkg/storage"
+)
+
+// KeyValueStore stores key-value pairs and maintains the range properties.
+type KeyValueStore struct {
+	dataWriter storage.ExternalFileWriter
+
+	rc        *rangePropertiesCollector
+	ctx       context.Context
+	offset    uint64
+	u64Buffer []byte
+}
+
+// NewKeyValueStore creates a new KeyValueStore. The data will be written to the
+// given dataWriter and range properties will be maintained in the given
+// rangePropertiesCollector.
+func NewKeyValueStore(
```

Comment:
Can we provide an function to clear all data stored in kv store？

Reply:
I'm not sure about the design from a higher perspective, seems NewKeyValueStore just wraps the writing functionality on external store, and clearing all data is still a functionality of external store itself.